common-types/src/main/scala/com/improving/app/common/errors/Validation.scala

```diff
@@ -1,85 +1,97 @@
 package com.improving.app.common.errors
 
-import com.improving.app.common.domain.{Address, CaPostalCodeImpl, Contact, MemberId, TenantId, UsPostalCodeImpl}
+import com.improving.app.common.domain.{Address, CaPostalCodeImpl, Contact, MemberId, OrganizationId, TenantId, UsPostalCodeImpl}
 
 object Validation {
 
-    type Validator[T] = T => Option[ValidationError]
+  type Validator[T] = T => Option[ValidationError]
 
-    def applyAllValidators[T](validators: Seq[Validator[T]]): Validator[T] =
-      (validatee: T) =>
-        validators.foldLeft[Option[ValidationError]](None)(
-          (maybeAlreadyError: Option[ValidationError], validator) =>
-            maybeAlreadyError.orElse(validator(validatee))
-        )
+  def applyAllValidators[T](validators: Seq[Validator[T]]): Validator[T] =
+    (validatee: T) =>
+      validators.foldLeft[Option[ValidationError]](None)(
+        (maybeAlreadyError: Option[ValidationError], validator) =>
+          maybeAlreadyError.orElse(validator(validatee))
+      )
 
-    def required[T]: (String, Validator[T]) => Validator[Option[T]] = (fieldName, validator) => {
-      opt =>
-        if (opt.isEmpty) {
-          Some(ValidationError(fieldName + " is missing."))
-        } else {
-          validator(opt.get)
-        }
+  def required[T]: (String, Validator[T]) => Validator[Option[T]] = (fieldName, validator) => {
+    opt =>
+      if (opt.isEmpty) {
+        Some(ValidationError(fieldName + " is missing."))
+      } else {
+        validator(opt.get)
+      }
+  }
+
+  def optional[T]: Validator[T] => Validator[Option[T]] = validator => opt => opt.flatMap(validator)
+
+  def skipEmpty(validator: Validation.Validator[String]): Validator[String] = str => {
+    if(str.isEmpty) {
+      None
+    } else {
+      validator(str)
     }
+  }
 
-    def optional[T]: Validator[T] => Validator[Option[T]] = validator => opt => opt.flatMap(validator)
+  val urlValidator: Validator[String] = url => {
+    None // TODO
+  }
 
-    def skipEmpty(validator: Validation.Validator[String]): Validator[String] = str => {
-      if(str.isEmpty) {
-        None
-      } else {
-        validator(str)
-      }
+  val tenantIdValidator: Validator[TenantId] = tenantId => {
```

Comment:
does this also check if the Optional TenantId is actually defined? I think if None is used by the user, it'll throw an exception

Reply:
likewise for the rest of the other functions that don't check for whether or not a None is possible

Reply:
the check for the `Option` happens in `required`. This validator is checking the id itself, given that it is present.

Reply:
Ah I see. Then I guess I'm not sure if we're handling that kind of exception well, since it's not very graceful as it errors out. Consider the grpcurl command: `grpcurl -plaintext -d '{"on_behalf_of": {"id": "establishingUser"}, "organization_info": {"name": "", "shortName": "", "tenant": {"id": "someTenantId"}, "is_public": true, "address": {"line1": "line1", "line2": "line2", "city": "city", "state_province": "stateProvince", "country": "country", "postal_code": {"ca_postal_code_message": "caPostalCode"}}, "url": "", "logo": ""}}' localhost:8082 com.improving.app.organization.api.OrganizationService/EstablishOrganization
`

Reply:
The above error is due to the OrganizationId being missing. I'm not clear on why it errors in that way, however.

Reply:
Ok, I see the issue here. It's actually on OrganizationServiceImpl.scala:42:
```
    val result = sharding.entityRefFor(Organization.TypeKey, in.organizationId.get.id)
```
We need to be doing some validation at that outer level, too.

Reply:
I think it just errors that way because it's not being handled as a StatusReply, rather, we're just letting the exception throw... right (I'm not too sure without looking too hard)?

Reply:
Just double checked and you're correct that I think the validation logic should be good. For that reason I'll resolve this. However what you mentioned was a good point and something needs to be done for that.